File: sendNotification.js
const admin = require('firebase-admin');
const serviceAccount = require('./skyhaven-project-7e626753c6cf.json');

// Initialize the Firebase app
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

// Define the message payload
const message = {
  notification: {
    title: 'Hello, World!',
    body: 'This is a test notification from FCM.'
  },
  token: 'BMlmHQtxNDDZBtOvsgXUfncF92BDk8SKIymFjJwMSEg9ho31b71A3paBzPsBIUII6137I22b5C5sMavVLqGdlsg' // Replace with the actual device token
};

// Send the notification
admin.messaging().send(message)
  .then((response) => {
    console.log('Successfully sent message:', response);
  })
  .catch((error) => {
    console.error('Error sending message:', error);
  });

File: send_Notification.py
import firebase_admin
from firebase_admin import credentials, messaging

# Initialize the Firebase app with the service account key
cred = credentials.Certificate("firebase-key.json")  # Ensure this file is in the same directory
firebase_admin.initialize_app(cred)

# Function to send a notification
def send_notification():
    # Create a message payload
    message = messaging.Message(
        notification=messaging.Notification(
            title="Test Notification",
            body="This is a test notification from Firebase HTTP v1",
        ),
        topic="airspace-alerts",  # Target the "airspace-alerts" topic
    )

    # Send the message
    response = messaging.send(message)
    print(f"Notification sent successfully! Response: {response}")

# Call the function
if __name__ == "__main__":
    send_notification()
